airbyte-server/src/main/java/io/airbyte/server/converters/SchemaConverter.java

```diff
@@ -24,107 +24,87 @@
-                  .withName(apiStream.getName())
-                  .withFields(persistenceFields)
-                  .withSupportedSyncModes(apiStream.getSupportedSyncModes()
-                      .stream()
-                      .map(e -> Enums.convertTo(e, io.airbyte.config.SyncMode.class))
-                      .collect(Collectors.toList()))
-                  .withSourceDefinedCursor(apiStream.getSourceDefinedCursor())
-                  .withDefaultCursorField(apiStream.getDefaultCursorField())
-                  // configurable
-                  .withSyncMode(Enums.convertTo(apiStream.getSyncMode(), io.airbyte.config.SyncMode.class))
-                  .withCursorField(apiStream.getCursorField())
-                  .withSelected(persistenceFields.stream().anyMatch(Field::getSelected));
-            })
-            .collect(Collectors.toList());
+  public static io.airbyte.protocol.models.AirbyteStream convertTo(final io.airbyte.api.model.AirbyteStream stream) {
+    return new io.airbyte.protocol.models.AirbyteStream()
+        .withName(stream.getName())
+        .withJsonSchema(stream.getJsonSchema())
+        .withSupportedSyncModes(Enums.convertListTo(stream.getSupportedSyncModes(), io.airbyte.protocol.models.SyncMode.class))
+        .withSourceDefinedCursor(stream.getSourceDefinedCursor())
+        .withDefaultCursorField(stream.getDefaultCursorField());
+  }
 
-    return new Schema().withStreams(persistenceStreams);
+  public static io.airbyte.api.model.AirbyteCatalog convertTo(final io.airbyte.protocol.models.AirbyteCatalog catalog) {
+    return new io.airbyte.api.model.AirbyteCatalog()
+        .streams(catalog.getStreams()
+            .stream()
+            .map(SchemaConverter::convertTo)
+            .map(s -> new io.airbyte.api.model.AirbyteStreamAndConfiguration()
+                .stream(s)
+                ._configuration(getDefaultAirbyteStreamConfiguration(s)))
+            .collect(Collectors.toList()));
   }
 
-  private static Field toPersistenceField(SourceSchemaField apiField) {
-    final Field field = new Field()
-        .withName(apiField.getName())
-        .withSelected(apiField.getSelected());
-    // todo (cgardens) - this is a hack to handle complex types (e.g. anyOf). they will not have a type
-    // field set, so we just call them "object". we need to revisit this ASAP.
-    if (apiField.getDataType() != null) {
-      field.withDataType(Enums.convertTo(apiField.getDataType(), DataType.class));
-    } else {
-      field.withDataType(DataType.OBJECT);
-    }
-    return field;
+  private static io.airbyte.api.model.AirbyteStreamConfiguration getDefaultAirbyteStreamConfiguration(io.airbyte.api.model.AirbyteStream s) {
+    io.airbyte.api.model.AirbyteStreamConfiguration result = new AirbyteStreamConfiguration()
+        .cleanedName(Names.toAlphanumericAndUnderscore(s.getName()))
+        .cursorField(s.getDefaultCursorField())
+        .selected(true);
```

Comment:
i don't like that these converters still aren't symmetrical. this is a fault in the BE data model (not in your implementation). we still lose information as the configuration comes back from the API. can you create an issue to make this symmetrical? i think ideally the backend model should look more like the FE. keeping all of the streams in the list but having a flag as to whether they are going to be synced. it's a bigger change though and we should not try to do it in this PR, but want to make sure we follow up.